app/controllers/works_controller.rb

```diff
@@ -0,0 +1,78 @@
+class WorksController < ApplicationController
+  def index
+    @works = Work.all
+  end
+
+  def show
+    work_id = params[:id]
+    @work = Work.find_by(id: work_id)
+    if @work.nil?
+      head :not_found
```

Comment:
Consider using a `redirect` instead of `head :not_found` so you can redirect the user to somewhere useful :)